### PR TITLE
systemd: use SIGUSR2 for shutdown, for LDM

### PR DESCRIPTION
**Pull request by philpennock**
### Changes proposed in this pull request:

 - tell systemd to use SIGUSR2 for shutdown, to trigger Lame Duck Mode in nats-server

### CODE REVIEW

**Review by wallyqs — APPROVED** [review]

**Review by derekcollison — APPROVED** [review]
LGTM - Question should we leave commented out though? This would change behavior yes?

**Comment by philpennock** [comment]
Someone would have to actually use this util/ script from the repo, as-is, automatically deploying it.  Do such setups exist?  AFAIK this is just a source for people to copy from during setup, if they so choose.

**Comment by derekcollison** [comment]
Oh yes they do. Without any edits.

File: util/nats-server-hardened.service [hunk 1]
```diff
@@ -9,6 +9,9 @@ ExecReload=/bin/kill -s HUP $MAINPID
 ExecStop=/bin/kill -s SIGINT $MAINPID
 User=nats
 Group=nats
+# The nats-server uses SIGUSR2 to trigger using Lame Duck Mode (LDM) shutdown
+KillSignal=SIGUSR2
+# You might want to adjust TimeoutStopSec too.
```
Comment (JulienVdG) [inline util/nats-server-hardened.service:14]: Hello, sorry for the late message, I only saw the change when browsing the release notes.

Shouldn't the ExecStop command also be updated to send SIGUSR2 ?

**Pull request merged**